src/main/java/net/snowflake/client/core/CredentialManager.java

```diff
@@ -92,18 +115,18 @@ void fillCachedMfaToken(SFLoginInput loginInput) throws SFException {
    * @param loginInput login input to attach token
    * @param credType credential type to retrieve
    */
-  synchronized void fillCachedCredential(SFLoginInput loginInput, String credType)
+  synchronized void fillCachedCredential(SFLoginInput loginInput, CachedCredentialType credType)
       throws SFException {
     if (secureStorageManager == null) {
       logMissingJnaJarForSecureLocalStorage();
       return;
     }
 
-    String cred = null;
+    String cred;
     try {
       cred =
           secureStorageManager.getCredential(
-              loginInput.getHostFromServerUrl(), loginInput.getUserName(), credType);
+              loginInput.getHostFromServerUrl(), loginInput.getUserName(), credType.getValue());
```

Comment:
I'm not sure if it is correct for oauth. Guessing from the code above `getHostFromServerUrl` returns snowflake host, right? For OAuth it should be IDP address. Otherwise we could accidentally leak tokens between IDPs.

Reply:
Refactored according to your suggestion.